CRM/Upgrade/Incremental/php/SixThree.php

```diff
@@ -29,6 +29,18 @@ class CRM_Upgrade_Incremental_php_SixThree extends CRM_Upgrade_Incremental_Base
    */
   public function upgrade_6_3_alpha1($rev): void {
     $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+
+    // Delete non-attachment rows in batches of 5000
+    $fileCount = CRM_Core_DAO::singleValueQuery('SELECT COUNT(*) FROM civicrm_entity_file WHERE entity_table LIKE "civicrm_value_%"');
```

Comment:
This might make it slower now, because of both count() and it's still running the LIKE query.

I did do some testing earlier on a medium size db and actually your original where it just does the delete without needing a batch seems to be fine. Sorry.

Reply:
IMO the batches will be slightly slower because of the extra count query up-front, but it's for good reason: a slightly slower upgrade is preferable to a crashing upgrade!